lib/drivers/asanteDriver.js

```diff
@@ -1204,8 +1179,48 @@ module.exports = function (config) {
     },
 
     uploadData: function (progress, data, cb) {
+      debug('Asante UploadData!');
+      data.stage = 'uploadData';
       progress(0);
-      console.log(data);
+
+      var postrecords = [], settings = null;
+      postrecords = asanteBuildSettingsRecord(data, postrecords);
+      if (!_.isEmpty(postrecords)) {
+        settings = postrecords[0];
+      }
+      postrecords = asanteBuildBolusRecords(data, postrecords);
+      postrecords = asanteBuildWizardRecords(data, postrecords);
+      postrecords = asanteBuildBasalRecords(data, postrecords);
+      postrecords = _.sortBy(postrecords, function(d) {
+        return d.time;
+      });
+
+      var simulator = asanteSimulatorMaker.make({settings: settings});
+
+      for (var i = 0; i < postrecords.length; ++i) {
+        var datum = postrecords[i];
+        switch (datum.type) {
+          case 'basal':
+            simulator.basal(datum);
+            break;
+          case 'bolus':
+            simulator.bolus(datum);
+            break;
+          case 'settings':
+            simulator.settings(datum);
+            break;
+          case 'smbg':
+            simulator.smbg(datum);
+            break;
+          case 'wizard':
+            simulator.wizard(datum);
+            break;
+          default:
```

Comment:
Given that all of these are the same signature, I think I would have done something like: 

````
var simfuncs = {
  basal: simulator.basal,
  bolus: simulator.bolus,
 ...
}; 

for (var i = 0; i < postrecords.length; ++i) {
        var datum = postrecords[i];
        if (!_.has(simfuncs, datum.type)) {
          // fail
        } else {
           simfuncs[datum.type](datum)
    }
}
````

In fact, looking at that now, I think the dispatcher belongs in the simulator.